mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapperTest.java

```diff
@@ -69,22 +82,363 @@
+                .thenReturn(Mono.error(new RuntimeException("mock exception")))
+                .thenCallRealMethod();
+
+            doQuietly(() -> testee.save(inboxRenamed));
+
+            SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+                softly.assertThat(testee.findMailboxById(inboxId))
+                    .isEqualTo(inbox);
+                softly.assertThat(testee.findMailboxByPath(inboxPath))
+                    .isEqualTo(inbox);
+                softly.assertThat(testee.findMailboxWithPathLike(inboxSearchQuery))
+                    .containsOnly(inbox);
+            }));
+        }
+
+        @Disabled("JAMES-3056 find by renamed name returns unexpected results")
+        @Test
+        void saveOnRenameThenFailToGetACLShouldBeConsistentWhenFindByRenamedInbox() throws Exception {
+            testee.save(inbox);
+
+            when(aclMapper.getACL(inboxId))
+                .thenReturn(Mono.error(new RuntimeException("mock exception")))
+                .thenCallRealMethod();
+
+            doQuietly(() -> testee.save(inboxRenamed));
+
+            SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+                softly.assertThatThrownBy(() -> testee.findMailboxByPath(inboxPathRenamed))
+                    .isInstanceOf(MailboxNotFoundException.class);
+                softly.assertThat(testee.findMailboxWithPathLike(inboxRenamedSearchQuery))
+                    .isEmpty();
+            }));
+        }
+
+        @Test
+        void saveOnRenameThenFailToRetrieveMailboxShouldBeConsistentWhenFindByInbox() throws Exception {
+            testee.save(inbox);
+
+            when(mailboxDAO.retrieveMailbox(inboxId))
+                .thenReturn(Mono.error(new RuntimeException("mock exception")))
+                .thenCallRealMethod();
+
+            doQuietly(() -> testee.save(inboxRenamed));
+
+            SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+                softly.assertThat(testee.findMailboxById(inboxId))
+                    .isEqualTo(inbox);
+                softly.assertThat(testee.findMailboxByPath(inboxPath))
+                    .isEqualTo(inbox);
+                softly.assertThat(testee.findMailboxWithPathLike(inboxSearchQuery))
+                    .containsOnly(inbox);
+            }));
+        }
+
+        @Disabled("JAMES-3056 find by renamed name returns unexpected results")
+        @Test
+        void saveOnRenameThenFailToRetrieveMailboxShouldBeConsistentWhenFindByRenamedInbox() throws Exception {
+            testee.save(inbox);
+
+            when(mailboxDAO.retrieveMailbox(inboxId))
+                .thenReturn(Mono.error(new RuntimeException("mock exception")))
+                .thenCallRealMethod();
+
+            doQuietly(() -> testee.save(inboxRenamed));
+
+            SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+                softly.assertThatThrownBy(() -> testee.findMailboxByPath(inboxPathRenamed))
```

Comment:
Am I able to retrieve the mailbox using INBOX path?

When I read the mailbox By ID what is its path?

Reply:
That's what I did in `saveOnRenameThenFailToRetrieveMailboxShouldBeConsistentWhenFindByInbox`

Reply:
can't all these assertions be grouped in the same soft-assertion ?

IMO as the inconsistency condition is the same, the soft-assertion should reflect the full 'big picture'

Reply:
I have a different opinion on that, disable the whole assertions while there are actually some are passed. If I review the code, I don't have the proper view on which assertion failed, which passed.